### PR TITLE
Make `CustomizableOperation` `Send` and `Sync`

### PR DESCRIPTION
**Pull request by ysaito1001**
## Motivation and Context
Fixes https://github.com/awslabs/smithy-rs/issues/2944

## Description
`CustomizableOperation` in the last two releases ([release-2023-08-01](https://github.com/awslabs/smithy-rs/releases/tag/release-2023-08-01) and [release-2023-08-22](https://github.com/awslabs/smithy-rs/releases/tag/release-2023-08-22)) broke backward-compatibility with respect to its auto traits. Specifically, it ceased to be `Send` and `Sync` because the struct contained a boxed trait object. This PR fix that issue, making `CustomizableOperation` `Send` and `Sync` again.

## Testing
- Added a Kotlin unit test to verify `CustomizableOperation` is `Send` and `Sync`.

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by Velfi — APPROVED** [review]
Thanks for writing the test.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/CustomizableOperationGenerator.kt [hunk 1]
```diff
@@ -185,14 +186,28 @@ class CustomizableOperationGenerator(
...
+                        interceptors: Vec<#{SharedInterceptor}>,
+                        runtime_plugins: Vec<#{SharedRuntimePlugin}>,
+                        _output: #{PhantomData}<T>,
+                        _error: #{PhantomData}<E>,
                     }
 
-                    impl<T, E> CustomizableOperation<T, E> {
+                    impl<T, E, B> CustomizableOperation<T, E, B> {
+                        /// Creates a new `CustomizableOperation` from `customizable_send`.
+                        pub fn new(customizable_send: B) -> Self {
```
Comment (jdisanti) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/CustomizableOperationGenerator.kt:200]: Should be `pub(crate)`?

Comment (ysaito1001) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/CustomizableOperationGenerator.kt:200]: Yes, thanks for catching!

**Review by jdisanti — APPROVED** [review]

**Comment by rcoh** [comment]
hmm, cargo semver-checks should have caught this 🤔

**Pull request merged**